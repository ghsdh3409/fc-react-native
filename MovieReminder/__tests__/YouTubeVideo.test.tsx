import React from 'react';
import YouTubeVideo from '../src/screens/MovieScreen/YouTubeVideo';
import renderer from 'react-test-renderer';
import { Linking, TouchableOpacity } from 'react-native';

describe('<YouTubeVideo />', () => {
  const youTubeKey = 'test_key';
  let testRenderer: renderer.ReactTestRenderer;
  beforeEach(() => {
    testRenderer = renderer.create(
      <YouTubeVideo title="Test Title" youTubeKey={youTubeKey} />,
    );
  });

  it('renders correctly', () => {
    const snapshot = testRenderer.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('opens url when it is pressed', () => {
    const spyFn = jest.spyOn(Linking, 'openURL');
    const touchableOpacity = testRenderer.root.findByType(TouchableOpacity);
    touchableOpacity.props.onPress();
    expect(spyFn).toHaveBeenCalledWith(
      `https://www.youtube.com/watch?v=${youTubeKey}`,
    );
  });
});
