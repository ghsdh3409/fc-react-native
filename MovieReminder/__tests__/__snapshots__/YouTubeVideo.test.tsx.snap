// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YouTubeVideo /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#ffffff",
      "borderRadius": 12,
      "opacity": 1,
      "overflow": "hidden",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://img.youtube.com/vi/test_key/0.jpg",
      }
    }
    style={
      Object {
        "height": 0,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 16,
        "padding": 12,
      }
    }
  >
    Test Title
  </Text>
</View>
`;
